### PR TITLE
Add additionalSize for the DisableOptData

### LINKED ISSUE #97461: ICorDebugFunction5::DisableOpimizations causes target app to be terminated on Linux
### Description

If a debugger uses ICorDebugFunction5::DisableOpimizations on Linux (and I am assuming macOS as well, but I didn't test) the target app will be terminated. Before terminated, the following message is printed:

```
Unknown debugger event type: 0x258
```

This message is from [this code](https://github.com/dotnet/runtime/blob/093fdabc28310bc8f60fcaecbcbf38a59d878056/src/coreclr/debug/shared/dbgtransportsession.cpp#L2504), and the event type is `DB_IPCE_DISABLE_OPTS`.

### Reproduction Steps

1. Create a new .NET 8 console application
2. Configure the debugger to disable Just My Code, and load symbols for System.Console (enable downloading symbol from the Microsoft symbol server)
3. Set a breakpoint on the call to Console.WriteLine
4. Step in

### Expected behavior

Console.WriteLine should be deoptimized and the step should complete

### Actual behavior

Target process is aborted

### Regression?

This is a new feature in .NET 8. So, it is a regression in that the VS debuggers will automatically use it if it is present and that causes the target to crash. 

### Known Workarounds

Don't use the new feature

### Configuration

I am using .NET Runtime 8.0.1 on Linux-x64

### Other information

_No response_

### PR DESCRIPTION
**Pull request by mikelle-rogers**
Add additional size to the buffer to read DisableOptData from memory before passing the IPCEvent.

### CODE REVIEW

File: src/coreclr/debug/shared/dbgtransportsession.cpp [hunk 1]
```diff
@@ -2499,6 +2499,8 @@ DWORD DbgTransportSession::GetEventSize(DebuggerIPCEvent *pEvent)
         break;
 
     case DB_IPCE_DISABLE_OPTS:
+        cbAdditionalSize = sizeof(pEvent->DisableOptData);
+        break;
 
     default:
         printf("Unknown debugger event type: 0x%x\n", (pEvent->type & DB_IPCE_TYPE_MASK));
```
Comment (davmason) [inline src/coreclr/debug/shared/dbgtransportsession.cpp:2506]: Can you change this to a stress log too?

Comment (hoyosjs) [inline src/coreclr/debug/shared/dbgtransportsession.cpp:2506]: Can you please also turn this into a stress log statement too?

**Review by hoyosjs — APPROVED** [review]

File: src/coreclr/debug/shared/dbgtransportsession.cpp (hunk 1 above)
Comment (hoyosjs) [inline src/coreclr/debug/shared/dbgtransportsession.cpp:2506]:
```suggestion
        STRESS_LOG1("Unknown debugger event type: 0x%x\n", (pEvent->type & DB_IPCE_TYPE_MASK));
```

File: src/coreclr/debug/shared/dbgtransportsession.cpp [hunk 2]
```diff
@@ -2499,9 +2499,11 @@ DWORD DbgTransportSession::GetEventSize(DebuggerIPCEvent *pEvent)
         break;
 
     case DB_IPCE_DISABLE_OPTS:
+        cbAdditionalSize = sizeof(pEvent->DisableOptData);
+        break;
 
     default:
-        printf("Unknown debugger event type: 0x%x\n", (pEvent->type & DB_IPCE_TYPE_MASK));
+        STRESS_LOG0(LF_CORDB, LL_INFO1000, "Unknown debugger event type: 0x%x\n", (pEvent->type & DB_IPCE_TYPE_MASK));
```
Comment (hoyosjs) [inline src/coreclr/debug/shared/dbgtransportsession.cpp:2506]:
```suggestion
        STRESS_LOG1(LF_CORDB, LL_INFO1000, "Unknown debugger event type: 0x%x\n", (pEvent->type & DB_IPCE_TYPE_MASK));
```

**Review by hoyosjs — CHANGES_REQUESTED** [review]

**Review by davmason — APPROVED** [review]
LGTM

**Review by hoyosjs — APPROVED** [review]

**Pull request merged**